test/tchannel.js

```diff
@@ -185,3 +187,22 @@ test('make socket: should throw for invalid destination', function t(assert) {
 //    'Should reject invalid destination');
 //  assert.end();
 //});
+
+
+test('make listen-on-fd socket: should work', function t(assert) {
+  var socket = new net.createServer();
+  socket.listen();
+
+  var server = new TChannel(serverOptions);
+  server.register('endpoint', function(_x, _y, _z, cb) { cb(null, 'ok'); });
+  server.listen(socket._handle.fd);
```

Comment:
this looks a bit weird. doesn't this need to be passed as: `{fd: socket._handle.fd}` ?